source/ucumLhcUtils.js

```diff
@@ -145,71 +145,158 @@ export class UcumLhcUtils {
   } // end validateUnitString
 
 
+
+/**
+ * @typedef {
+ *   'normal', 
+ *   'mol->mass', 
+ *   'mass->mol', 
+ *   'eq->mass', 
+ *   'mass->eq', 
+ *   'eq->mol', 
+ *   'mol->eq', 
+ *   'mol<->mol', 
+ *   'eq<->eq'
+ * } ConversionType
+ */
+
+  /**
+  * Detects the type of conversion between two units.
+  *
+  * @param {Object} fromUnit - The unit to convert from.
+  * @param {Object} toUnit - The unit to convert to.
+  * @returns {ConversionType} conversionType - The type of conversion as a string. Possible values are:
+  * 'normal', 'mol->mass', 'mass->mol', 'eq->mass', 'mass->eq', 'eq->mol', 'mol->eq', 'mol<->mol', 'eq<->eq'.
+  */
+  detectConversionType(fromUnit, toUnit) {
+    /** @type {ConversionType} */
+    let conversionType = 'normal';
+
+    // if neither unit is not a molar unit, return 'normal'
+    if (!(fromUnit.isMolarUnit() || toUnit.isMolarUnit())) {
+      return conversionType;
+    }
+     
+    // Otherwise, handle mol <-> eq <-> mass conversions
+    if (fromUnit.isMolarUnit() && toUnit.isMolarUnit() && !fromUnit.isEquivalentUnit() && !toUnit.isEquivalentUnit()) {
+      conversionType = 'mol<->mol';
+    } else if (fromUnit.isEquivalentUnit() && toUnit.isEquivalentUnit()) {
+      conversionType = 'eq<->eq';
+    }
+    // handle eq <-> mol/mass conversions
+    else if (fromUnit.isEquivalentUnit() || toUnit.isEquivalentUnit()) {
+      if (fromUnit.isEquivalentUnit() && toUnit.isMolarUnit()) {
+        conversionType = 'eq->mol';
+      } else if (fromUnit.isMolarUnit() && toUnit.isEquivalentUnit()) {
+        conversionType = 'mol->eq';
+      } else if (fromUnit.isEquivalentUnit() && !toUnit.isMolarUnit()) {
+        conversionType = 'eq->mass';
+      } else if (!fromUnit.isMolarUnit() && toUnit.isEquivalentUnit()) {
+        conversionType = 'mass->eq';
+      }
+    } 
+    // handle mol <-> mass conversions
+    else {
```

Comment:
This is fair, I'll update the place of that check.

There is one footgun with the isEquiv and isMolar functions that prevents me from using the code snippet as stated:

```
if(from.isEquiv && to.isEquiv)
else if(from.isEquiv)
else if(to.isEquiv)
else ...
```
While isEquiv will only return true for eq units, isMolar will return true for both eq and molar units.

I will need to continue to check both to and from for equiv:

```
if(from.isEquiv && to.isEquiv)
else if(from.isEquiv && to.isMolar)
else if(to.isEquiv && from.isMolar)
else if(from.isEquiv)
else if(to.isEquiv)
else ...
```

However, moving to one flat if/else if statement/workflow makes sense. I'll also add a comment that explains why the additional isMolar checks are there

Reply:
sounds good. I'm not requesting additional change heres - would it work if you check isMolar under the if(from.isEquiv) and if(to.isEquiv) clause?
Whatever you feel comfortable with. The original code was fine, too, just a bit difficult to follow given the combinations.

Reply:
I do like the idea of adding the check under the if, especially since it'll cut down lines of code without impacting clarity. Even though its not a request, I'll update the style of the code to the following:

```js
    // handle eq <-> mol/mass conversions
    else if (fromUnit.isEquivalentUnit()) {
      conversionType = toUnit.isMolarUnit() ? 'eq->mol' : 'eq->mass';
    } else if (toUnit.isEquivalentUnit()) {
      conversionType = fromUnit.isMolarUnit() ? 'mol->eq' : 'mass->eq';
    }
```